clustering/polus-hdbscan-clustering-plugin/src/main.py

```diff
@@ -0,0 +1,168 @@
+        
+    Returns:
+        The path to directory, list of names of the subdirectories in dirpath (if any) and the filenames of .csv files.
+        
+    """
+    list_of_files = [os.path.join(dirpath, file_name)
+                     for dirpath, dirnames, files in os.walk(csv_directory)
+                     for file_name in fnmatch.filter(files, '*.csv')]
+    return list_of_files
+
+
+def clustering(data: np.ndarray, min_cluster_size: int, increment_outlier_id: bool) -> np.ndarray:
+    """Cluster data using HDBSCAN.
+    
+    Args:
+        data (array): Data that need to be clustered.
+        min_cluster_size (int): Smallest size grouping that should be considered as a cluster.
+        increment_outlier_id (bool) : Increment outlier ID to unity.
+        
+    Returns:
+        Cluster labels for each row of data.
+    """
+    clusters = hdbscan.HDBSCAN(min_cluster_size=min_cluster_size).fit(data)
+    labels = clusters.labels_.flatten().astype(np.uint16) + 1
+    labels = labels + 1 if increment_outlier_id else labels
+
+    return labels
+    
+   
+# Setup the argument parsing
+def main():
+    logger.info("Parsing arguments...")
+    parser = argparse.ArgumentParser(prog='main', description='HDBSCAN clustering plugin')
+    parser.add_argument('--inpDir', dest='inpDir', type=str,
+                        help='Input collection-Data need to be clustered', required=True)
+    parser.add_argument('--groupingPattern', dest='groupingPattern', type=str,
+                        help='Regular expression to group rows. Clustering will be applied across capture groups.', required=False)
+    parser.add_argument('--averageGroups', dest='averageGroups', type=str,
+                        help='Whether to average data across groups. Requires capture groups.', default='false', required=False)
+    parser.add_argument('--labelCol', dest='labelCol', type=str,
+                        help='Name of column containing labels. Required only for grouping operations.', required=False)
+    parser.add_argument('--minClusterSize', dest='minClusterSize', type=int,
+                        help='Minimum cluster size', required=True)
+    parser.add_argument('--incrementOutlierId', dest='incrementOutlierId', type=str,
+                        help='Increments outlier ID to 1.', default='false', required=False)
+    parser.add_argument('--outDir', dest='outDir', type=str,
+                        help='Output collection', required=True)
+    
+    # Parse the arguments.
+    args = parser.parse_args()
+    
+    # Path to csvfile directory.
+    inpDir = args.inpDir
+    logger.info('inpDir = {}'.format(inpDir))
+
+    # Regular expression for grouping.
+    grouping_pattern = args.groupingPattern
+    logger.info('grouping_pattern = {}'.format(grouping_pattern))
+
+    # Whether to average data for each group.
+    avg_groups = args.averageGroups.lower() != 'false'
+    logger.info('avg_groups = {}'.format(avg_groups))
```

Comment:
If we are rewriting this, argument parsing should be done under an `if __name__=="__main__"` block and passed to the `main` function.